Feat/poke stats

seburiticav opened the pull request:
## Pull Request Description

**Summary:** Added the full stats system with per-level multipliers and updated the encapsulation of some values.
<!-- Brief description of changes -->


### Type of Change
- [ ] 🐛 Bug fix
- [X] ✨ New feature  
- [ ] 💥 Breaking change
- [ ] 📝 Documentation
- [ ] 🔧 Refactoring

### Changes Made
- Added stats logic
- Added leveling up stat scaling 
- Updated the class diagram with the changes made on pokemon.py
- Added testing for pokemon stats
### Testing
- [ ] Manual testing completed

### Screenshots (if applicable)

### Checklist
- [X] Code self-reviewed
- [ ] Code commented where needed
- [ ] No new warnings/errors
- [X] Documentation updated
- [X] Ready for review

seburiticav commented:
Por cierto profe, hice mas cambios pero debido a un error a la hora de leer el mensaje los hice en type inheritance y no se marcan en este PR, sin embargo si es necesario puedo mandar una captura con todos los cambios que realice con el tiempo

felipeg17 commented on pokemon/pokemon.py:
```diff
@@ -1,15 +1,17 @@
     def __init__(
             self,
-            name: str,
+            pokemonname: str,
```
snake case - pokemon_name

felipeg17 commented on pokemon/pokemon.py:
```diff
@@ -1,15 +1,17 @@
 import os
 import pdb
+import pandas as pd
```
Es necesario agregarlo a los requerimientos - requirements.txt

felipeg17 commented on pokemon/pokemon.py:
```diff
@@ -1,15 +1,17 @@
 from pprint import pprint
 
 class Pokemon:
+    csv_path = 'https://raw.githubusercontent.com/seburiticav/poke-repo/refs/heads/feat/poke-type-inheritance/pokemon/First30Pokemons.csv'
```
por qué llama a un archivo en la web, si igual está local, puede crear una carpeta utils, y ahi poner First30Pokemons, ojo ser consistente con el naming.

felipeg17 commented on pokemon/pokemon.py:
```diff
@@ -60,7 +73,34 @@ def receive_attack(self, attack_type):
+        self.base_attack = int(row['Attack'].values[0])
+        self.base_defense = int(row['Defense'].values[0])
+        self.base_spattack = int(row['Sp. Attack'].values[0])
+        self.base_spdefense = int(row['Sp. Defense'].values[0])
```
snake case base_sp_defense

felipeg17 commented on pokemon/pokemon.py:
```diff
@@ -60,7 +73,34 @@ def receive_attack(self, attack_type):
+        self.spattack = self.base_spattack
+        self.spdefense = self.base_spdefense
+        self.speed = self.base_speed
+    def combatstats(self, accuracy = "100%", evasion = "100%"):
```
snake casing

felipeg17 commented on pokemon/First30Pokemons.csv:
```diff
@@ -0,0 +1,31 @@
+Pokedex Number,Pokemon Name,HP,Attack,Defense ,Sp. Atk,Sp. Def,Speed
```
usar una convencion estandar en los nombres de las columnas

seburiticav replied:
a que se refiere con una convencion estandar profe?

felipeg17 replied:
a que las columas en el archivo tienen una forma de nombrarse distinta a la que se usa dentro del codigo, e.g. "Pokemon Name" vs "pokemon_name"

felipeg17 reviewed (changes requested):
Ajustar el tema de naming con snake case
Ajustar ruta del archivo .csv

seburiticav commented:
## Pull Request Description

**Summary:** Fixed variable names with snake case and created a new folder for utils like the csv with the first 30 pokemons of the pokedex
<!-- Brief description of changes -->

### Type of Change
- [ ] 🐛 Bug fix
- [X] ✨ New feature  
- [ ] 💥 Breaking change
- [ ] 📝 Documentation
- [X] 🔧 Refactoring

### Changes Made
- Added Utils folder with the .csv file
- Added pandas to requeriments.txt
- Fixed some variable names with snake case
- Updated csv_path with the new path of the .csv (utils/First30Pokemons.csv)

### Testing
- [ ] Manual testing completed

### Screenshots (if applicable)

### Checklist
- [X] Code self-reviewed
- [ ] Code commented where needed
- [X] No new warnings/errors
- [X] Documentation updated
- [X] Ready for review

felipeg17 reviewed (approved):
Solo falta ajustar el tema del nombre de las columnas, pero lo voy a aprobar
